sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/BoundAttribute.scala

```diff
@@ -53,7 +53,7 @@ object BindReferences extends Logging {
             sys.error(s"Couldn't find $a in ${input.mkString("[", ",", "]")}")
           }
         } else {
-          BoundReference(ordinal, a.dataType, a.nullable)
+          BoundReference(ordinal, input(ordinal).dataType, a.nullable)
```

Comment:
before this, we use ' val ordinal = input.indexWhere(_.exprId == a.exprId)' to find the AttributeReference which is equal to 'a', but the dataType in 'a' is compele, the dataType in input(ordinal) has been cutted in file 'ParquetTableOperations', you can see the 'output' property in case class ParquetTableScan